finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4DispatcherBase.scala

```diff
@@ -105,6 +105,14 @@ trait Netty4DispatcherBase[SendMsg <: Message, RecvMsg <: Message] {
         log.debug(e, "[%s] dispatcher closed", prefix)
         Future.exception(e)
 
+      case Throw(e: ChannelClosedException)
+      // if all streams have already been closed, then this just means that
+      // the client failed to send a GOAWAY frame...
+      if closed.get || streams.keys.asScala.forall { _ <= closedId.get } =>
```

Comment:
If I'm understanding correctly, streams should be removed from `streams` when they are closed.  So I think this can just be `closed.get || streams.isEmpty`.